src/components/UserActions/UserActions.ts

```diff
@@ -306,6 +315,16 @@ export class UserActions extends Component {
             }
         });
     }
+
+    private inferResultListLayout(): ResultLayoutType {
+        const resultLists = this.root.querySelectorAll<HTMLElement>(`${Component.computeSelectorForType(ResultList.ID)}:not(.coveo-hidden)`);
+        const resultListLayoutTypes = [ResultLayoutType.CARD, ResultLayoutType.TABLE, ResultLayoutType.LIST] as string[];
+
+        if (resultLists.length > 0 && resultListLayoutTypes.indexOf(resultLists[0].dataset.layout) !== -1) {
```

Comment:
You are checking the first result list that is visible. Maybe it is far-fetched, but suppose you have two result lists visible at the same time. The first one has the LIST template and the second one has the TABLE template. Since you are checking the layout for the first visible list, it will still append the "Viewed by Customer" element in the result list with the TABLE template.

Instead, can you check the layout on the first `CoveoResultList` parent?

Reply:
Sadly no, that was my first plan but sadly it's not possible. the dom note given in the handler is not yet attached to the dom.